### PR TITLE
Iterators: Add Python runtime/execution engine stub

### PR DESCRIPTION
**Pull request by ingomueller-net**
This PR adds utility functions that allow passing pandas DataFrames into jitted code. It depends on and currently includes #600 and #579.

### CODE REVIEW

File: experimental/iterators/test/python/dialects/iterators/dialect.py [hunk 1]
```diff
@@ -98,3 +103,33 @@ def testEndToEnd():
...
+        "iterators.sink"(%stream) : (!iterators.stream<!struct_type>) -> ()
+        return
+      }
+      ''')
+  pm = PassManager.parse(
+      'convert-iterators-to-llvm,convert-memref-to-llvm,convert-func-to-llvm,'
+      'reconcile-unrealized-casts,convert-scf-to-cf,convert-cf-to-llvm')
+  pm.run(mod)
+
+  data = np.array([(0, 3), (1, 4), (2, 5)], dtype=[('a', 'i4'), ('b', 'i8')])
```
Comment (nicolasvasilache) [inline experimental/iterators/test/python/dialects/iterators/dialect.py:127]: Plz comment the format here, as it is non-standard.

File: experimental/iterators/test/python/dialects/iterators/dialect.py [hunk 2]
```diff
@@ -98,3 +103,33 @@ def testEndToEnd():
...
+
+
+@run
+# CHECK-LABEL: TEST: testEndToEndWithInput
+def testEndToEndWithInput():
+  mod = Module.parse('''
+      !struct_type = !llvm.struct<(i32,i64)>
+      func.func @main(%input: !tabular.tabular_view<i32,i64>)
+          attributes { llvm.emit_c_interface } {
+        %stream = "iterators.tabular_view_to_stream"(%input)
```
Comment (nicolasvasilache) [inline experimental/iterators/test/python/dialects/iterators/dialect.py:115]: can we use nicer assembly everywhere plz?

**Review by nicolasvasilache — APPROVED** [review]

File: experimental/iterators/test/python/dialects/iterators/dialect.py (hunk 2 above)
Comment (ingomueller-net) [reply experimental/iterators/test/python/dialects/iterators/dialect.py]: Applied in latest commit.

File: experimental/iterators/test/python/dialects/iterators/dialect.py (hunk 1 above)
Comment (ingomueller-net) [reply experimental/iterators/test/python/dialects/iterators/dialect.py]: Improved in latest commit.

**Pull request merged**